File: flood.java
class Solution {
    int m,n,row,col;
    int[][] dirs={{0,-1},{0,1},{-1,0},{1,0}};
    public int[][] floodFill(int[][] image, int sr, int sc, int newColor) {

        int startColor=image[sr][sc];

        return helper(image,sr,sc,newColor,startColor);

    }

    public int[][] helper(int[][] image,int sr,int sc,int newColor,int startColor){


        m=image.length;
        n=image[0].length;
        if(sr>-1 && sr<m && sc>-1 && sc<n && image[sr][sc]!=newColor && image[sr][sc]==startColor){
            for(int[] dir:dirs){

                image[sr][sc]=newColor;


                row=sr+dir[0];
                col=sc+dir[1];
                image= helper(image,row,col,newColor,startColor);
            }
        }
        return image;

    }
}

File: flood.py
# // Time Complexity :O(mn)
# // Space Complexity : O(mn)
# // Did this code successfully run on Leetcode :Yes
# // Any problem you faced while coding this :No

# you pop the current color at desired location given and add the locationindex to the queue. Loop until the queue is empty to find the four neighbouring nodes
# of current position and if they are within the bounds and if the color at the nehigbours are old color- then add them to queue and change clors to new color
class Solution(object):
    def floodFill(self, image, sr, sc, color):
        """
        :type image: List[List[int]]
        :type sr: int
        :type sc: int
        :type color: int
        :rtype: List[List[int]]
        """
        m=len(image)
        n=len(image[0])
        if image[sr][sc]==color:
            return image
        old_color=image[sr][sc]
        q=[]
        dirs=[[0,1],[0,-1],[1,0],[-1,0]]
        q.append([sr,sc])
        image[sr][sc]=color
        while(len(q)>0):
            curr=q.pop(0)
            for dir in dirs:
                nr=curr[0]+dir[0]
                nc=curr[1]+dir[1]
                if(nr>=0 and nc>=0 and nr<m and nc<n and image[nr][nc]==old_color):
                    q.append([nr,nc])
                    image[nr][nc]=color

        return image